research/delf/delf/python/datasets/tuples_dataset.py

```diff
@@ -13,51 +13,51 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-"""Tuple dataset based on the Radenovic et al. ECCV16: CNN image retrieval
-learns from BoW.
+"""Tuple dataset module.
 
+Based on the Radenovic et al. ECCV16: CNN image retrieval learns from BoW.
 For more information refer to https://arxiv.org/abs/1604.02426.
 """
 
 import os
 import pickle
 
-from absl import flags
-import matplotlib.pyplot as plt
 import numpy as np
 import tensorflow as tf
 
-from delf.python.datasets import generic_dataset
 from delf.python.datasets import utils as image_loading_utils
 from delf.python.training import global_features_utils
-
-FLAGS = flags.FLAGS
+from delf.python.training.model import global_model
 
 
 class TuplesDataset():
   """Data loader that loads training and validation tuples.
 
+  After initialization, the function create_epoch_tuples() should be called to
+  create the dataset tuples. After that, the dataset can be iterated through
+  using next() function.
   Tuples are based on Radenovic et al. ECCV16 work: CNN image retrieval
   learns from BoW. For more information refer to
   https://arxiv.org/abs/1604.02426.
   """
 
-  def __init__(self, name, mode, data_root, imsize=None, nnum=5, qsize=2000,
-               poolsize=20000, loader=image_loading_utils.default_loader,
-               ims_root=None):
+  def __init__(self, name, mode, data_root, imsize=None, num_negatives=5,
+               num_queries=2000, poolsize=20000,
```

Comment:
nit: can you please rename all occurrences of "poolsize" by "pool_size"? It would be agree better with the naming convention we're using ("num_negatives", "num_queries", etc)